server/src/test/java/org/opensearch/index/query/ExistsQueryBuilderTests.java

```diff
@@ -89,9 +89,14 @@ protected ExistsQueryBuilder doCreateTestQueryBuilder() {
         }
         // also sometimes test wildcard patterns
         if (randomBoolean()) {
-            if (randomBoolean()) {
-                fieldPattern = fieldPattern + "*";
-            }
+            fieldPattern = fieldPattern + "*";
+        }
+        // Avoid patterns that would match the derived field "raw.derived_keyword"
+        // This includes "raw", "raw*", "raw.xxx", "r*", "ra*", etc.
+        if (fieldPattern.equals("raw")
+            || fieldPattern.startsWith("raw")
+            || (fieldPattern.contains("*") && "raw.derived_keyword".startsWith(fieldPattern.replace("*", "")))) {
```

Comment:
Can the pattern matching checks can be moved before the wild-card logic block, so that you don't have to worry about replacing `"*"` with `""`?

In case you match the restricted patterns, override early and then when you add `*` to `fieldPattern`, that should be safe to do so.

What do you think?

Reply:
Agreed, fixed.